src/Cli/ConfigGenerator.cs

```diff
@@ -918,7 +938,150 @@ private static bool TryUpdateConfiguredCacheValues(
+                if (updatedProviderValue != null)
+                {
+                    updatedValue = updatedProviderValue?.ToString() ?? nameof(EasyAuthType.StaticWebApps);
+                    AuthenticationOptions AuthOptions;
+                    if (updatedHostOptions?.Authentication == null)
+                    {
+                        AuthOptions = new(Provider: (string)updatedValue);
+                    }
+                    else
+                    {
+                        AuthOptions = updatedHostOptions.Authentication with { Provider = (string)updatedValue };
+                    }
+
+                    updatedHostOptions = updatedHostOptions! with { Authentication = AuthOptions };
+                    _logger.LogInformation("Updated RuntimeConfig with Runtime.Host.Authentication.Provider as '{updatedValue}'", updatedValue);
+                }
+
+                // Runtime.Host.Authentication.Jwt.Audience
+                updatedValue = options?.RuntimeHostAuthenticationJwtAudience;
+                if (updatedValue != null)
+                {
+                    JwtOptions jwtOptions;
+                    AuthenticationOptions AuthOptions;
+                    if (updatedHostOptions?.Authentication == null || updatedHostOptions.Authentication?.Jwt == null)
+                    {
+                        jwtOptions = new(Audience: (string)updatedValue, null);
+                    }
+                    else
+                    {
+                        jwtOptions = updatedHostOptions.Authentication.Jwt with { Audience = (string)updatedValue };
+                    }
+
+                    if (updatedHostOptions?.Authentication == null)
+                    {
+                        AuthOptions = new(Jwt: jwtOptions);
+                    }
+                    else
+                    {
+                        AuthOptions = updatedHostOptions.Authentication with { Jwt = jwtOptions };
+                    }
+
+                    updatedHostOptions = updatedHostOptions! with { Authentication = AuthOptions };
+                    _logger.LogInformation("Updated RuntimeConfig with Runtime.Host.Authentication.Jwt.Audience as '{updatedValue}'", updatedValue);
+                }
+
+                // Runtime.Host.Authentication.Jwt.Issuer
+                updatedValue = options?.RuntimeHostAuthenticationJwtIssuer;
+                if (updatedValue != null)
+                {
+                    JwtOptions jwtOptions;
+                    AuthenticationOptions AuthOptions;
+                    if (updatedHostOptions?.Authentication == null || updatedHostOptions.Authentication?.Jwt == null)
+                    {
+                        jwtOptions = new(null, Issuer: (string)updatedValue);
+                    }
+                    else
+                    {
+                        jwtOptions = updatedHostOptions.Authentication.Jwt with { Issuer = (string)updatedValue };
```

Comment:
IIRC, Authentication.Jwt and can have audience and issuer values only when Authentication.Provider is Jwt. Can we validate before updating audience/issuer values that Provider is Jwt? If not, fail the update?